gui/dig.lua

```diff
@@ -190,9 +198,39 @@ function GenericOptionsPanel:init()
                 {
                     label = "Ramp",
                     value = "r",
-                } },
+                }
+            },
             disabled = false,
             show_tooltip = true,
+            on_change = function(new, old) self.dig_panel:updateLayout() end,
+        },
+        widgets.CycleHotkeyLabel {
+            view_id = "stairs_subtype",
+            key = "CUSTOM_G",
+            key_back = "CUSTOM_SHIFT_G",
+            label = "Stair Type: ",
```

Comment:
I think you might need two of these (both defaulting to Auto) so the player can define what they want both at the top and the bottom

Reply:
or maybe a single one with the combinations that make sense? Auto/Auto, Down/Up, UpDown/Up, Down/UpDown, UpDown/UpDown

Reply:
I think I'll just go with two single-direction cycle labels.